src/components/localization/InstallationLanguage.jsx

```diff
@@ -294,22 +297,66 @@ class LanguageSelector extends React.Component {
     }
 }
 
+const KeyboardSelector = ({ keyboards, selectedKeyboard, setKeyboard }) => {
+    useEffect(() => {
+        // Check if the selected keyboard matches the current keyboards
+        if (keyboards.length > 0) {
+            const selectedLayoutId = selectedKeyboard?.split(":")[0];
+            const isKeyboardValid = keyboards.some(
+                ({ layoutId }) => layoutId === selectedLayoutId
+            );
+
+            // Reset to default if current selection is invalid
+            if (!isKeyboardValid) {
+                setKeyboard(`${keyboards[0].layoutId}:${keyboards[0].variantId || ""}`);
+            }
+        }
+    }, [keyboards, selectedKeyboard, setKeyboard]);
+
+    const handleChange = (event) => {
+        const { value } = event.target;
+        setKeyboard(value);
+    };
+
+    const selectedValue =
+        selectedKeyboard ||
+        (keyboards.length > 0 ? `${keyboards[0].layoutId}:${keyboards[0].variantId || ""}` : "");
+
+    return (
+        <FormSelect
+          value={selectedValue}
+          onChange={handleChange}
+          aria-label="Select Keyboard Layout"
```

Comment:
You dont need aria label, just add an id to the element, and supply that to fieldId of the FormGroup.